pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -695,19 +713,31 @@ func (r *ReconcileAWSPrivateLink) reconcileVPCEndpointService(awsClient *awsClie
 	for _, allowed := range permResp.AllowedPrincipals {
 		oldPerms.Insert(aws.StringValue(allowed.Principal))
 	}
-	desriredPerms := sets.NewString(aws.StringValue(stsResp.Arn))
+	// desiredPerms is the set of Allowed Principals that will be configured for the cluster's VPC Endpoint Service.
+	// desiredPerms only contains the IAM entity used by Hive (defaultARN) by default but may contain additional Allowed Principal
+	// ARNs as configured within cd.Spec.Platform.AWS.PrivateLink.AdditionalAllowedPrincipals.
+	defaultARN := aws.StringValue(stsResp.Arn)
+	desiredPerms := sets.NewString(defaultARN)
+	if allowedPrincipals := cd.Spec.Platform.AWS.PrivateLink.AdditionalAllowedPrincipals; allowedPrincipals != nil {
+		for _, allowedPrincipal := range *allowedPrincipals {
+			desiredPerms.Insert(allowedPrincipal)
+		}
```

Comment:
`Insert` is variadic, so you should be able to do this without the loop.
```suggestion
		desiredPerms.Insert(*allowedPrincipals...)
```